homeassistant/components/google_photos/media_source.py

```diff
@@ -0,0 +1,283 @@
+"""Media source for Google Photos."""
+
+from dataclasses import dataclass
+import logging
+from typing import Any, cast
+
+from homeassistant.components.media_player import MediaClass, MediaType
+from homeassistant.components.media_source import (
+    BrowseError,
+    BrowseMediaSource,
+    MediaSource,
+    MediaSourceItem,
+    PlayMedia,
+)
+from homeassistant.core import HomeAssistant
+
+from . import GooglePhotosConfigEntry
+from .const import DOMAIN
+from .exceptions import GooglePhotosApiError
+
+_LOGGER = logging.getLogger(__name__)
+
+# Media Sources do not support paging, so we only show a subset of recent
+# photos when displaying the users library. We fetch a minimum of 50 photos
+# unless we run out, but in pages of 100 at a time given sometimes responses
+# may only contain a handful of items Fetches at least 50 photos.
+MAX_PHOTOS = 50
+PAGE_SIZE = 100
+
+THUMBNAIL_SIZE = 256
+LARGE_IMAGE_SIZE = 2048
+
+
+# Markers for parts of PhotosIdentifier url pattern.
+# The PhotosIdentifier can be in the following forms:
+#  config-entry-id
+#  config-entry-id/a/album-media-id
+#  config-entry-id/p/photo-media-id
+#
+# The album-media-id can contain special reserved folder names for use by
+# this integration for virtual folders like the `recent` album.
+PHOTO_SOURCE_IDENTIFIER_PHOTO = "p"
+PHOTO_SOURCE_IDENTIFIER_ALBUM = "a"
+
+# Currently supports a single album of recent photos
+RECENT_PHOTOS_ALBUM = "recent"
+RECENT_PHOTOS_TITLE = "Recent Photos"
+
+
+@dataclass
+class PhotosIdentifier:
+    """Google Photos item identifier in a media source URL."""
+
+    config_entry_id: str
+    """Identifies the account for the media item."""
+
+    album_media_id: str | None = None
+    """Identifies the album contents to show.
+
+    Not present at the same time as `photo_media_id`.
+    """
+
+    photo_media_id: str | None = None
+    """Identifies an indiviidual photo or video.
```

Comment:
individual

Reply:
Fixed in https://github.com/home-assistant/core/pull/124957